src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala

```diff
@@ -70,6 +72,11 @@ object Pipeline {
         analysis.keywords.nonEmpty
       }
 
+      def hasBlacklistedTerms(details: Details): Boolean = {
```

Comment:
Ideally, this method should be added to the `Analyzer` interface, and this implementation should exist in `AnalysisDefaults`. The type of the details parameter there should be changed to accept ExtendedDetails[T] to give each concrete `Analyzer` the chance to override it, and `blacklist` will need to be added as a parameter as well.